Add filter for Closure functions

akoryak opened the pull request:
#### What does this PR do?
Add filter for Closure functions

#### Checklist
- [ x ] Pull request contains a clear definition of changes
- [ x ] Tests (either unit, integration, or acceptance) written and passing
- [ x ] Relevant documentation produced and/or updated

chrisryan commented on src/Filter/Closures.php:
```diff
@@ -0,0 +1,41 @@
+     * The return value is \Closure, as expected by the \TraderInteractive\Filterer class.
+     *
+     * @param mixed $value       the value to filter to a boolean
+     * @param bool  $allowNull   Set to true if NULL values are allowed. The filtered result of a NULL value is NULL
+     * @return bool|null the filtered $value
```
```suggestion
     * @param bool  $allowNull   Set to true if NULL values are allowed. The filtered result of a NULL value is NULL
     *
     * @return bool|null the filtered $value
```

chadicus reviewed (commented):
I'm not sure this is the right library for the closures filter class, but i'll defer to others.
Be sure to update the README.md file with instructions and usage examples

akoryak commented:
> I'm not sure this is the right library for the closures filter class, but i'll defer to others. Be sure to update the README.md file with instructions and usage examples

1) updated the documentation
2) It doesn't have much sense to me to create new git repository with ~90% duplication for just one simple filter function. Second reason I why I added it here is that I don't see any use case for this closure filter be used standalone

chadicus commented on src/Filter/Closures.php:
```diff
@@ -0,0 +1,39 @@
+     * @param mixed $value       the value to filter to a closure function
+     * @param bool  $allowNull   Set to true if NULL values are allowed. The filtered result of a NULL value is NULL
+     *
+     * @return bool|null the filtered $value
```
```suggestion
     * @return Closure|null the filtered $value
```

chadicus commented on src/Filter/Closures.php:
```diff
@@ -0,0 +1,39 @@
+     * @param mixed $value       the value to filter to a closure function
+     * @param bool  $allowNull   Set to true if NULL values are allowed. The filtered result of a NULL value is NULL
+     *
+     * @return Closure|null the filtered $value
```
```suggestion
     * @return \Closure|null the filtered $value
```
